src/index.js

```diff
@@ -74,7 +81,39 @@ function handlebars(options) {
   const Handlebars = options.handlebars.module || require('handlebars');
   const ImportScanner = require('./ImportScanner')(Handlebars);
 
+  const hbsImport = `import Handlebars from '${escapePath(options.handlebars.id)}';\n`;
+
+  const wrapTemplateDefinition =
+    options.helpersPureInitialize === 'expression'
+      ? (defineTemplate, initExpr) => defineTemplate((expr) => `(${initExpr},${expr})`)
+      : options.helpersPureInitialize
+      ? (defineTemplate, initExpr) =>
+          defineTemplate((expr) => `(function() {${initExpr};return ${expr};})()`)
+      : (defineTemplate, initExpr) => `${initExpr};\n${defineTemplate()}`;
+
+  // Support `helpers` being singular or plural.
+  const helpers = asArrayOr(options.helpers, null);
+
   return {
+    resolveId: (id) => (helpers && id === INTERNAL_INIT_ID ? id : undefined),
+
+    load(id) {
+      if (!helpers || id !== INTERNAL_INIT_ID) return;
+
+      let body = hbsImport;
+      body += '';
+
+      const initExpr = helpers.map((helperPath, i) => {
+        const ref = `Helpers${i}`;
+        body += `import ${ref} from '${escapePath(helperPath)}';\n`;
+        return `  ${ref}.__initialized || (${ref}(Handlebars), ${ref}.__initialized = true);\n`;
```

Comment:
is this meant to be an indent?

Reply:
Yep! It's inside a function and this library has tended to produce nicely-ish formatted code.